core/src/View/Kokkos_ViewCtor.hpp

```diff
@@ -213,11 +213,16 @@ struct ViewCtorProp : public ViewCtorProp<void, P>... {
   using execution_space = typename var_execution_space::type;
   using pointer_type    = typename var_pointer::type;
 
-  /*  Copy from a matching argument list.
-   *  Requires  std::is_same< P , ViewCtorProp< void , Args >::value ...
-   */
-  template <typename... Args>
-  inline ViewCtorProp(Args const &...args) : ViewCtorProp<void, P>(args)... {}
+  // Construct from a matching argument list.
+  //
+  // Note that if P is empty, this constructor is the default constructor.
+  // On the other hand, if P is not empty, the constraint implies that
+  // the default constructor cannot be called.
+  template <typename... Args,
+            typename = std::enable_if_t<std::conjunction_v<
+                std::is_constructible<ViewCtorProp<void, P>, Args &&>...>>>
+  ViewCtorProp(Args &&...args)
+      : ViewCtorProp<void, P>(std::forward<Args>(args))... {}
```

Comment:
It seems this doesn't work with the Intel Classic compiler
```suggestion
// The Intel Classic Compiler struggles deducing the variadic template parameters with the SFINAE template parameter 
#ifdef KOKKOS_COMPILER_INTEL
 template <typename... Args>
#else
  template <typename... Args,
            typename = std::enable_if_t<std::conjunction_v<
                std::is_constructible<ViewCtorProp<void, P>, Args &&>...>>>
#endif
 ViewCtorProp(Args &&...args)
      : ViewCtorProp<void, P>(std::forward<Args>(args))... {}
```

Reply:
Indeed. We found that it can be solved by using the `view_ctor_prop_args` typedef. So we preferred that solution over using a preprocessor conditional.